### ISSUE TITLE
Vim Various Text Objects

### ISSUE THREAD

**Issue by ovidius72** [zed-industries/zed#15293]
### Check for existing issues

- [X] Completed

### Describe the feature

There is a plugin that allows to use 30 extra text objects. https://github.com/chrisgrieser/nvim-various-textobjs
Some of them are really useful and I use them all the days.
I was wondering if it is possibile to add them in zed (or at least some of them).

Thanks.

**Comment by Avi-D-coder** [comment]
Adding support for tree-sitter and or lsp text object would also be great.

**Comment by AlbertMarashi** [comment]
Here i thought something like this would be implemented

**Comment by AlbertMarashi** [comment]
How hard would it be to add support for a zed editor command that allows anyone to run any kind of arbitrary tree-sitter queries?

```json
"(": ["editor::scm_query", "...SCM QUERY HERE..."]
```

**Comment by 0x2CA** [comment]
@ConradIrwin https://github.com/nvim-treesitter/nvim-treesitter-textobjects 
I wonder if this would be a good start for the zed plugin to extend vim?
This feature sounds like it might be a good fit for

**Comment by ConradIrwin** [comment]
Yeah. That said, much of it is also something we should have by default.

**Comment by kunsttyv** [comment]
@ConradIrwin treesitter textobjects in Zed would be such a killer feature, `[ x` is already quite useful, as limited as it is. I am a Python programmer and being able to do modal editing on AST objects would be extremely powerful, almost like writing Lisp in Emacs.

**Comment by ConradIrwin** [comment]
We recently shipped support for classes, methods and comments; which seem to be the "90% reward for 10% effort" set. We also have arguments already.

Feel free to open other issues if there are other objects you constantly miss.
